### DISCUSSION TITLE
Prevent unessesary work when Overriding variables in the called task

### DISCUSSION THREAD

**Discussion by ameershira** [go-task/task#494] [Q&A] [answered]
Hi,

I created a common build task that gets called from other tasks. The other tasks override the common build task variables. The common build task is informed about the source and generated files. The common build task's source & generated files are changed for every task via the variable overrides. 

When running the other tasks twice sequentially, i expect that the second time running the tasks would state that the tasks are up-to-date, but instead it behaves as if the other tasks all affect each other and every task thus gets run and rebuilt again.

 Any assistance would be appreciated

**Comment by ghost** [comment, marked as answer]
Hi @andreynering, can I perhaps get some feedback on this? do you need more info?
Thanks in advance.

**Reply by andreynering** [reply]
Hi @ameershira-uman,

Sorry, too many open issues, too little time. I sometimes take time to answer each one.

If I understood correctly what you're trying to accomplish, I think that having a custom `label:` would do the trick. Please see: https://github.com/go-task/task/issues/394#issuecomment-860255357

We may improve that soon. If you have opinions, feel free to add your feedback in the issue.

If I misunderstood your question, please add an example to clarify what you mean.

**Reply by ameershira** [reply]
Thanks for the feedback @andreynering. No worries. Yes i used the custom label exactly as you stated. I'm going install the latest version tonight and retest. I will provide further feedback.

**Reply by ameershira** [reply]
@andreynering i upgraded and i don't see the issue i had previously.
